library/ssl_msg.c

```diff
@@ -645,7 +643,7 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
             return( MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );
         }
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
-        if( transform->minor_ver >= MBEDTLS_SSL_MINOR_VERSION_1 )
+        if( transform->minor_ver == MBEDTLS_SSL_MINOR_VERSION_3 )
```

Comment:
This breaks support MBEDTLS_SSL_MINOR_VERSION_4.
Do not change the `>=`
Please fix other similar changes.

Reply:
It seems to me that `transform->minor_ver >= MBEDTLS_SSL_MINOR_VERSION_3` is always true at this point, so perhaps we should remove the check entirely and save some code size (and indentation level)? Same for other places where `>=` is used. (If there are places with `==` of course the check should stay, for the sake of 1.3.) (Also, obviously when we're checking a version sent by the peer rather than stored internally, the check also needs to stay.)

Reply:
Yes, now it looks like overzealousness or lack of confidence ;-)
I think it can be removed without compromising security.